anomalib/models/dfkde/model.py

```diff
@@ -20,13 +20,71 @@
 import torchvision
 from omegaconf.dictconfig import DictConfig
 from omegaconf.listconfig import ListConfig
-from torch import Tensor
+from torch import Tensor, nn
 
 from anomalib.models.components import AnomalyModule, FeatureExtractor
 
 from .normality_model import NormalityModel
 
 
+class DfkdeModel(nn.Module):
+    """DFKDR model.
+
+    Args:
+        backbone (nn.Module): Feature extraction backbone.
+        filter_count (int): Number of filters.
+        threshold_steepness (float): Threshold steepness for normality model.
+        threshold_offset (float): Threshold offset for normality model.
+    """
+
+    def __init__(
+        self, backbone: nn.Module, filter_count: int, threshold_steepness: float, threshold_offset: float
+    ) -> None:
+        super().__init__()
+        self.feature_extractor = FeatureExtractor(backbone=backbone(pretrained=True), layers=["avgpool"]).eval()
+
+        self.normality_model = NormalityModel(
+            filter_count=filter_count,
+            threshold_steepness=threshold_steepness,
+            threshold_offset=threshold_offset,
+        )
```

Comment:
This would probably work, but I feel like the model would still need another refactoring. Ideally `DfkdeModel` should be the normality model that contains feature_extractor, pca and kde modules. I guess this would be scope of another PR, but something to keep in mind.

Reply:
Alternatively, there could be a better naming than `NormalityModel` if we want to keep `NormalityModel`. For instance, in DFM, there `SingleClassGaussian`. Again, another PR

Reply:
Created a ticket for it https://github.com/openvinotoolkit/anomalib/issues/129